WowPacketParserModule.V8_0_1_27101/Parsers/IslandHandler.cs

```diff
@@ -0,0 +1,92 @@
+
+namespace WowPacketParserModule.V8_0_1_27101.Parsers
+{
+    public static class IslandHandler
+    {
+        [Parser(Opcode.SMSG_ISLAND_AZERITE_XP_GAIN)]
+        public static void HandleIslandAzeriteXpGain(Packet packet)
+        {            
+            packet.ReadUInt32("XpGain");
+            packet.ReadPackedGuid128("PlayerGuid");
+            packet.ReadPackedGuid128("SourceGuid");
+            packet.ReadUInt32("SourceID");
+        }
+
+        public static void ReadIslandPlayerData(Packet packet, params object[] idx)
+        {
+            packet.ReadPackedGuid128("PlayerGuid");
+            var itemEquipedCount = packet.ReadInt32("ItemEquipedCount", idx);
+            packet.ResetBitReader();
+            var PlayerNameLen = packet.ReadBits(6);
+
+            packet.ReadByteE<Gender>("Gender", idx);
+            packet.ReadByte("Skin", idx);
+            packet.ReadByte("HairColor", idx);
+            packet.ReadByte("HairStyle", idx);
+            packet.ReadByte("FacialStyle", idx);
+            packet.ReadByte("Face", idx);
+            packet.ReadByteE<Race>("Race", idx);
+            packet.ReadByteE<Class>("Class", idx);
+            for (int i = 0; i < 3; i++)
+                packet.ReadByte("CustomDisplay", idx, i);
+
+            packet.ReadWoWString("PlayerName", PlayerNameLen, idx);
+
+            for (int j = 0; j < itemEquipedCount; j++)
+                ReadIslandItemEquiped(packet, idx, j, "ItemEquiped");
+        }
+
+        public static void ReadIslandItemEquiped(Packet packet, params object[] idx)
+        {
+            packet.ReadPackedGuid128("ItemGuid", idx);
+            packet.ReadByte("EquipmentSlot", idx);
+            var AzeriteTierSetcount = packet.ReadInt32("AzeriteTierSetCount", idx);
+            for (int i = 0; i < AzeriteTierSetcount; i++)
+                packet.ReadInt32("AzeritePowerId", idx, i);
+
+            Substructures.ItemHandler.ReadItemInstance(packet, idx);
+
+            packet.ResetBitReader();
+            packet.ReadBit("Usable", idx);
+
+            var enchantsCount = packet.ReadBits("EnchantsCount", 4, idx);
+            var gemsCount = packet.ReadBits("GemsCount", 2, idx);
+            for (int j = 0; j < gemsCount; j++)
+            {
+                packet.ReadByte("Slot", idx, j);
+                Substructures.ItemHandler.ReadItemInstance(packet, idx, j);
+            }
+            for (int k = 0; k < enchantsCount; k++)
+            {
+                packet.ReadUInt32("Id", idx, k);
+                packet.ReadByte("Index", idx, k);
+            }
+        }
+
+        [Parser(Opcode.SMSG_ISLAND_COMPLETED)]
+        public static void HandleIslandCompleted(Packet packet)
+        {
+            packet.ReadInt32("MapID");
+            packet.ReadInt32("Winner");
+            var PlayerCount = packet.ReadUInt32("GroupTeamSize");
+            for (int j = 0; j < PlayerCount; j++)
+                ReadIslandPlayerData(packet, j, "PlayerData");
```

Comment:
Use CharacterHandler.ReadPlayerModelDisplayInfo here (yes, the structure is shared in client between SMSG_ISLAND_COMPLETED and SMSG_INSPECT_RESULT

Reply:
Ok I gonna take a review, ty

Reply:
did it